lib/server/data.js

```diff
@@ -0,0 +1,103 @@
+KdDataLayer = function(args) {
+  this._serverId = args.serverId;
+  this._docID = {
+    'client': 'clientAuthorizedSessions',
+    'remote': 'remoteAuthorizedSessions',
+  }
+
+  this.configColl = new Mongo.Collection('__kdconfig');
+  this.timeEventsColl = new Mongo.Collection('__kdtimeevents');
+  this.timeEventsColl._createCappedCollection(52428800); // 50 MB
+  this.tracesColl = new Mongo.Collection('__kdtraces');
+  this.tracesColl._createCappedCollection(52428800); // 50 MB
+
+  // set TTL Index for expire accessTokens and authSessions
+  this.configColl._ensureIndex({expires: 1}, {expireAfterSeconds: 3600});
+};
+
+KdDataLayer.prototype.registerAccessToken = function(accessToken) {
+  var dateNow = new Date();
+  var expiryDate = dateNow.setDate(dateNow.getDate() + 1);
+
+  this.configColl.insert({
+    type: 'accessTokens',
+    token: accessToken,
+    expires: expiryDate
+  });
+};
+
+KdDataLayer.prototype.isValidToken = function(accessToken) {
+  return !!this.configColl.findOne({
+    type: 'accessTokens',
+    token: accessToken
+  });
+};
+
+KdDataLayer.prototype.registerSession = function(type, sessionId) {
+  var dateNow = new Date();
+  var expiryDate = dateNow.setDate(dateNow.getDate() + 1);
+
+  this.configColl.insert({
+    type: this._docID[type],
+    session: sessionId,
+    expires: expiryDate
+  });
+};
+
+KdDataLayer.prototype.isValidSession = function(type, sessionId) {
+  return !!this.configColl.findOne({
+    type: this._docID[type], 
+    session: sessionId
+  });
+};
+
+KdDataLayer.prototype.unregisterSession = function(type, sessionId) {
+  this.configColl.remove({
+    type: this._docID[type],
+    session: sessionId
+  });
+};
+
+KdDataLayer.prototype.setListenersCount = function(val) {
+  this.configColl.update(
+    { _id: 'listenersCount' },
+    { $inc: {count: val}},
+    { upsert: true }
+  );
+};
+
+KdDataLayer.prototype.getListenersCount = function() {
+  var config = this.configColl.findOne({_id: 'listenersCount'});
+  var timelineCount = (config && config.count) ? config.count : 0;
+  return timelineCount;
+};
+
+KdDataLayer.prototype.setTimeEvent = function(data) {
+  this.timeEventsColl.insert(data);
```

Comment:
Do this operation via the node API. So, it won't any overhead.
Do it like this:

```
this.configColl._rawCollection().insert(data, function(err) {
  if(err) {
    console.error(error);
  }
})
```